source/guide_rocketchat.rst

```diff
@@ -0,0 +1,305 @@
+
+::
+
+ [isabell@stardust ~]$ uberspace tools version show node
+ Using 'Node.js' version: '12'
+ [isabell@stardust ~]$
+
+We'll also need :lab:`MongoDB <guide_mongodb>`, so follow the MongoDB guide and come back when it's running.
+
+Check your chat URL setup:
+
+::
+
+ [isabell@stardust ~]$ uberspace web domain list
+ isabell.uber.space
+ [isabell@stardust ~]$
+
+Set up the backend:
+
+::
+
+ [isabell@stardust ~]$ uberspace web backend set / --http --port 3000
+ Set backend for / to port 3000; please make sure something is listening!
+ You can always check the status of your backend using "uberspace web backend list".
+ [isabell@stardust ~]$
+
+Installation
+============
+
+Download the latest Rocket.Chat release:
+
+::
+
+ [isabell@stardust ~]$ curl -L https://releases.rocket.chat/latest/download -o rocket.chat.tgz
+ curl -L https://releases.rocket.chat/latest/download -o rocket.chat.tgz
+   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
+                                  Dload  Upload   Total   Spent    Left  Speed
+ 100   223  100   223    0     0    333      0 --:--:-- --:--:-- --:--:--   333
+ 100  146M  100  146M    0     0  13.0M      0  0:00:11  0:00:11 --:--:-- 14.0M
+ [isabell@stardust ~]$ 
+
+And then extract the archive, use ``--strip-components=1`` to remove the ``bundle`` prefix from the path:
+
+::
+
+ [isabell@stardust ~]$ mkdir ~/rocket.chat
+ [isabell@stardust ~]$ tar -xzf rocket.chat.tgz -C ~/rocket.chat --strip-components=1
+ 
+You can delete the archive now:
+
+::
+
+ [isabell@stardust ~]$ rm rocket.chat.tgz
+ [isabell@stardust ~]$
+
+Now we install the required dependencies:
+
+::
+
+ [isabell@stardust ~]$ cd ~/rocket.chat/programs/server
+ [isabell@stardust server]$ npm install
+ [...]
+ very long npm chatter
+ [...]
+ [isabell@stardust server]$ cd ~
+ [isabell@stardust ~]$
+ 
+
+Configuration
+=============
+
+Configure MongoDB
+-----------------
+
+First stop MongoDB
+
+::
+
+ [isabell@stardust ~]$ supervisorctl stop mongodb
+ mongodb: stopped
+ [isabell@stardust ~]$
+
+Update the daemon configuration file ``~/etc/services.d/mongodb.ini``, add the option ``--replSet rs01``:
+
+.. code-block:: ini
+
+ [program:mongodb]
+ command=mongod
+   --dbpath %(ENV_HOME)s/mongodb
+   --bind_ip 127.0.0.1
+   --auth
+   --unixSocketPrefix %(ENV_HOME)s/mongodb
+   --replSet rs01
+ autostart=yes
+ autorestart=yes
+
+This is required because Rocket.Chat uses Meteor Oplog Tailing for performance improvements (see `the docs <https://rocket.chat/docs/installation/manual-installation/mongo-replicas/>`_ for further information). 
+
+Then tell supervisord to update and start the service:
+
+::
+
+ [isabell@stardust ~]$ supervisorctl update
+ mongodb: stopped
+ mongodb: updated process group
+ [isabell@stardust ~]$ 
+
+Now initiate the replica set:
+
+::
+
+ [isabell@stardust ~]$ mongo --username <username>_mongoroot --eval "printjson(rs.initiate())"
```

Comment:
The mongodb guide suggest to use the uberspace username as `<username>`, so I think we can replace all `<username>_mongoroot` with `${USER}_mongoroot` to make the guide easier to follow.